cmd/packer-plugin-check/main.go

```diff
@@ -0,0 +1,177 @@
+)
+
+// Usage is a replacement usage function for the flags package.
+func Usage() {
+	fmt.Fprintf(os.Stderr, "Usage of "+packerPluginCheck+":\n")
+	fmt.Fprintf(os.Stderr, "\t"+packerPluginCheck+" [flags]\n")
+	fmt.Fprintf(os.Stderr, "Flags:\n")
+	flag.PrintDefaults()
+}
+
+func main() {
+	log.SetFlags(0)
+	log.SetPrefix(packerPluginCheck + ": ")
+	flag.Usage = Usage
+	flag.Parse()
+
+	if flag.NFlag() == 0 {
+		flag.Usage()
+		os.Exit(2)
+	}
+
+	if *docs {
+		if err := checkDocumentation(); err != nil {
+			fmt.Printf(err.Error())
+			os.Exit(2)
+		}
+		fmt.Printf("Plugin succesfully passed docs check.\n")
+	}
+
+	if len(*load) > 0 {
+		if err := checkPluginName(*load); err != nil {
+			fmt.Printf(err.Error())
+			os.Exit(2)
+		}
+		if err := discoverAndLoad(); err != nil {
+			fmt.Printf(err.Error())
+			os.Exit(2)
+		}
+		fmt.Printf("Plugin succesfully passed compatibility check.\n")
+	}
+}
+
+// checkDocumentation looks for the presence of a docs folder with mdx files inside.
+// It is not possible to predict the number of mdx files for a given plugin.
+// Because of that, finding one file inside de folder is enough to template the knowledge of docs generation.
+func checkDocumentation() error {
+	// TODO: this should be updated once we have defined what's going to be for plguin's docs
+	wd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	docsDir := wd + "/docs"
+	stat, err := os.Stat(docsDir)
+	if err != nil {
+		return fmt.Errorf("could not find docs folter: %s", err.Error())
+	}
+	if !stat.IsDir() {
+		return fmt.Errorf("expecting docs do be a directory of mdx files")
+	}
+
+	var mdxFound bool
+	_ = filepath.Walk(docsDir, func(path string, info os.FileInfo, err error) error {
+		if !info.IsDir() && filepath.Ext(path) == ".mdx" {
+			mdxFound = true
+			return io.EOF
+		}
+		return nil
+	})
+
+	if mdxFound {
+		fmt.Printf("a mdx file was found inside the docs folder\n")
+		return nil
+	}
+	return fmt.Errorf("no docs files found, make sure to have the docs in place before releasing")
+}
+
+// checkPluginName checks for the possible valid names for a plugin, packer-plugin-* or packer-[builder|provisioner|post-processor]-*.
+// If the name is prefixed with `packer-[builder|provisioner|post-processor]-`, packer won't be able to install it,
+// therefore a WARNING will be shown.
+func checkPluginName(name string) error {
+	if strings.HasPrefix(name, "packer-plugin-") {
+		return nil
+	}
+	if strings.HasPrefix(name, "packer-builder-") ||
+		strings.HasPrefix(name, "packer-provisioner-") ||
+		strings.HasPrefix(name, "packer-post-processor-") {
+		fmt.Printf("\n[WARNING] Plugin is named with old prefix `packer-[builder|provisioner|post-processor]-{name})`. " +
+			"These will be detected but Packer cannot install them automatically. " +
+			"The plguin must be a multiplugin named as packer-plugin-{name} to become available through `packer init` command.\n")
```

Comment:
```suggestion
			"The plugin must be a multiplugin named as packer-plugin-{name} to become available through `packer init` command.\n")
```

Reply:
Why do I always put the letters in the wrong order 🤦  "plguin"